var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -271,6 +271,9 @@ class Gcc(AutotoolsPackage, GNUMirrorPackage):
     patch('glibc-2.31-libsanitizer-2.patch', when='@8.1.0:8.3.0,9.0.0:9.2.0')
     patch('glibc-2.31-libsanitizer-2-gcc-6.patch', when='@5.3.0:5.5.0,6.1.0:6.5.0')
     patch('glibc-2.31-libsanitizer-2-gcc-7.patch', when='@7.1.0:7.5.0')
+    patch('https://gcc.gnu.org/git/?p=gcc.git;a=patch;h=2b40941d23b1570cdd90083b58fa0f66aa58c86e', sha256='b48e48736062e64a6da7cbe7e21a6c1c89422d1f49ef547c73b479a3f3f4935f', when='@:9.3.0')
+    patch('https://github.com/gcc-mirror/gcc/commit/745dae5923aba02982563481d75a21595df22ff8.patch', sha256='7686d3b3a7ad0dc3f44a4141067133d2c708ea2cba503f684389e7bbb3bfa81a', when='@10.1.0:11.1.0')
```

Comment:
Might be my fault as I was suggesting the second patch above. But I agree to be consistent with the two patches.

Reply:
Hi @alalazo, @michaelkuhn

I have updated the script with the requested changes. I checked with spack patch, as well as, compiling some specific versions requiring (or not) either of the patches.

Reply:
Hi,

Sorry, I have to apologize. I have just realized versions like 7.1.0 or 8.1.0 do actually need the patch for compilation. The problem is the patching process fails, e.g., for gcc@7.1.0:
`==> Fetching https://mirror.spack.io/_source-cache/archive/8a/8a8136c235f64c6fef69cac0d73a46a1a09bb250776a050aec8f9fc880bebc17.tar.bz2
==> Fetching https://gcc.gnu.org/git/?p=gcc.git;a=patch;h=2b40941d23b1570cdd90083b58fa0f66aa58c86e
==> Applied patch /opt/spack/var/spack/repos/builtin/packages/gcc/glibc-2.31-libsanitizer-1.patch
==> Applied patch /opt/spack/var/spack/repos/builtin/packages/gcc/glibc-2.31-libsanitizer-2-gcc-7.patch
1 out of 3 hunks FAILED -- saving rejects to file libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cc.rej
==> Patch https://gcc.gnu.org/git/?p=gcc.git;a=patch;h=2b40941d23b1570cdd90083b58fa0f66aa58c86e failed.
==> Error: Command exited with status 1:
'/usr/bin/patch' '-s' '-p' '1' '-i' '/home/castanedamedina/.spack/stage/spack-stage-_q0btyzf/?p=gcc.git;a=patch;h=2b40941d23b1570cdd90083b58fa0f66aa58c86e' '-d' '.'
`

Reply:
I have seen similar problems with GCC patches in the past since even minor releases seem to move stuff around. I would say the two patches you added are good enough, people will hopefully not use very old versions of GCC on very new versions of Linux.